jwql/shared_tasks/shared_tasks.py

```diff
@@ -0,0 +1,516 @@
+        else:
+            model.ramp_fit.save_opt = False
+            fitopt_output = None
+            run_fitopt = False
+    else:
+        model.ramp_fit.skip = True
+        pipe_output = None
+        fitopt_output = None
+        run_slope = False
+        run_fitopt = False
+
+    # Call the pipeline if any of the files at the requested calibration
+    # states are not present in the output directory
+    logging.info("*****CELERY: Running save_jump pipeline")
+    if run_jump or (ramp_fit and run_slope) or (save_fitopt and run_fitopt):
+        model.run(datamodel)
+    else:
+        print(("Files with all requested calibration states for {} already present in "
+               "output directory. Skipping pipeline call.".format(input_file)))
+    
+    calibrated_files = glob(uncal_file.replace("_uncal.fits", "*"))
+    copy_files(calibrated_files, output_dir)
+
+    logging.info("*****CELERY: Finished pipeline")
+    return jump_output, pipe_output, fitopt_output
+
+
+def run_pipeline(input_file, in_ext, ext_or_exts, instrument, jump_pipe=False):
+    """Convenience function for using the ``run_calwebb_detector1`` function on a data 
+    file, including the following steps:
+    
+    - Lock the file ID so that no other calibration happens at the same time
+    - Copy the input (raw) file to the (central storage) transfer location
+    - Call the ``run_calwebb_detector1`` task
+    - For the extension (or extensions) (where by "extension" we mean 'uncal' or 'refpix'
+      or 'jump' rather than something like '.fits') requested, copy the files from the 
+      outgoing transfer location to the same directory as the input file
+    - Delete the input file from the transfer location
+    - Delete the output files from the transfer location
+    
+    It will then return what it was given – either a single file+path or a list of 
+    files+paths, depending on what ``out_exts`` was provided as.
+
+    Parameters
+    ----------
+    input_file : str
+        Name of fits file to run on the pipeline
+
+    ext_or_exts : str or list-of-str
+        The requested output calibrated files
+    
+    instrument : str
+        Name of the instrument being calibrated
+    
+    jump_pipe : bool
+        Whether the detector1 jump pipeline is being used (e.g. the bad pixel monitor)
+
+    Returns
+    -------
+    file_or_files : str or list-of-str
+        Name (or names) of the result file(s), including path(s)
+    """
+    logging.info("Pipeline Call for {} requesting {}".format(input_file, ext_or_exts))
+    config = get_config()
+    send_path = os.path.join(config["transfer_dir"], "incoming")
+    ensure_dir_exists(send_path)
+    receive_path = os.path.join(config["transfer_dir"], "outgoing")
+    ensure_dir_exists(receive_path)
+    
+    input_path, input_name = os.path.split(input_file)
+    logging.info("\tPath is {}, file is {}".format(input_path, input_name))
+
+    if "uncal" not in in_ext:
```

Comment:
From code review: Remove this `if/else` block.

Reply:
Done.